include/stl2/detail/iterator/move_iterator.hpp

```diff
@@ -90,6 +91,33 @@ STL2_OPEN_NAMESPACE {
 				static_cast<void>(++current_)
 			)
 
+			// Experimental support for move_iterator post-increment
+			using __post_inc_t = decltype(current_++);
+			Readable{R}
+			struct __proxy {
+				R __tmp;
+				decltype(auto) operator*()
+				STL2_NOEXCEPT_RETURN(
+					__stl2::iter_move(__tmp)
+				)
+				friend decltype(auto) iter_move(const __proxy& that)
+				STL2_NOEXCEPT_RETURN(
+					__stl2::iter_move(that.__tmp)
+				)
+			};
+
+			STL2_CONSTEXPR_EXT cursor post_increment()
+			noexcept(noexcept(cursor{cursor{__stl2::exchange(current_, __stl2::next(current_))}}))
+			requires Same<__post_inc_t, I>() {
+				return cursor{__stl2::exchange(current_, __stl2::next(current_))};
```

Comment:
Ah. So this picks the `post_increment` overload that is implemented in terms of _pre_-increment, and then we end up seeing a stomped value, is that right? OK, what if we nuked that overload of `post_increment()` and only kept the one with the proxy that is implemented in terms of _post_-fix increment?